server/controllers/users.js

```diff
@@ -138,20 +147,133 @@ class UserController {
             as: 'profile',
             attributes: ['firstname', 'lastname', 'bio', 'avatar', 'location']
-          },
-        ],
+          }
+        ]
       });
 
-      return res.status(200)
-        .send({
-          status: 'success',
-          message: 'Users and corresponding profiles retrieved successfully',
-          payload: users
-        });
+      return Response.success(res, 200, users, 'Users and corresponding profiles retrieved successfully');
     } catch (error) {
       next(error);
     }
   }
+
+  /**
+   * Forgot Password
+   * @async
+   * @param  {object} req - Request object
+   * @param {object} res - Response object
+   * @param {object} next
+   * @return {json} Returns json object
+   * @static
+   */
+  static async forgotPassword(req, res, next) {
+    try {
+      const { email } = await validateForgotPassword(req.body);
+
+      const user = await User.findOne({ where: { email } });
+
+      if (!user) return Response.error(res, 404, 'User does not exist');
+
+      // Check if the account is active
+      if (!user.active) return Response.error(res, 403, 'Your account is not verified');
+
+      const token = await UserController.updateToken(user);
+
+      sendForgotPasswordMail(token, email, user.get().username);
+
+      return Response.success(res, 200, { }, 'A reset token has been sent to your email address');
+    } catch (err) {
+      if (err.isJoi && err.name === 'ValidationError') {
+        return res.status(400).json({
+          status: 400,
+          errors: validationResponse(err)
+        });
+      }
+      next(err);
+    }
+  }
+
+  /**
+   * Update token
+   * @async
+   * @param {object} user
+   * @return {string} Returns token string
+   * @static
+   */
+  static async updateToken(user) {
+    const token = randomString({ length: 40 });
+    const resetDetails = {
+      resetPasswordToken: token,
+      resetPasswordExpiry: Date.now() + Number(process.env.RESET_TOKEN_EXPIRE
+        || 75600000) // 1 day from now
+    };
+
+    const userResetToken = await user.getResetToken();
+    if (userResetToken) {
+      await userResetToken.update(resetDetails);
+    } else {
+      await user.createResetToken(resetDetails);
+    }
+    return token;
+  }
+
+  /**
+   * Reset Password
+   * @async
+   * @param  {object} req - Request object
+   * @param {object} res - Response object
+   * @param {object} next
+   * @return {json} Returns json object
+   * @static
+   */
+  static async resetPassword(req, res, next) {
+    try {
+      // Verify the password and confirm password
+      const newPassword = await validatePasswordReset(req.body);
+
+      const { password } = newPassword;
+      const { token, email } = req.query;
+
+      if (!token || !email) return Response.error(res, 400, 'Please use a valid reset link');
+
+      // Get the user
+      const user = await User.findOne({ where: { email } });
+      if (!user) return Response.error(res, 404, 'User does not exist');
```

Comment:
Your current implementation is a sort of security breach (You are giving a would be attacker information about if an email exists on our system or not)
consider using this instead
```suggestion
      if (!user) return Response.error(res, 400, 'Please use a valid reset link');
```

Reply:
Makes sense. Thanks